lib/attrsets.nix

```diff
@@ -872,6 +878,62 @@ rec {
     else
       [];
 
+
+  /**
+    Recursively collect sets that verify a given predicate named `pred` from the arbitrary nesting of
+    sets and lists `attrs`. The recursion is stopped when the predicate is verified.
+
+    In other words, it behaves like [`lib.attrsets.collect`](#function-library-lib.attrsets.collect) but also recurses on lists.
+
+    # Inputs
+
+    `pred`
+
+    : Given an attribute's value, determine if recursion should stop.
+
+    `attrs`
+
+    : The arbitrary nesting of sets and lists to recursively collect from.
+
+    # Type
+
+    ```
+    collect :: (AttrSet -> Bool) -> AttrSet -> [x]
```

Comment:
> (regarding Any, ...

and if that doesn't feel right, maybe that's because a function like this would:

```
concatFromData :: (Any -> ListOf a) -> Any -> ListOf a
```

but the predicate function does not produce anything specific beyond a bool, so

```
:: (Any -> Bool) -> Any -> ListOf a
```

but that's just

```
:: (Any -> Bool) -> Any -> ListOf Any
```